src/scripts/main.js

```diff
@@ -1 +1,29 @@
 'use strict';
+
+const mainImg = document.getElementById('largeImg');
+const galleryList = document.getElementById('thumbs');
+
+const images = document.querySelectorAll('.gallery__thumb');
+const links = document.querySelectorAll('.list-item__link');
+
+galleryList.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  const checkClick = e.target.closest('.list-item__link');
+
+  if (!checkClick) {
+    return;
+  }
+
+  images.forEach((image) => {
```

Comment:
you already have the correct link in `checkClick` variable - there is no need to iterate through other images/links to find the correct one

P.S. rename checkClick variable

Reply:
You didn't rename the variable - you've deleted it but kept the loops

try this:

```
const imageLink = e.target.closest('.list-item__link');

if (!imageLink) {
  return;
}

mainImg.src = imageLink.href;
```

If necessary you can check if target is a link (tag name or class) then use e.target.href
But it should work without it

Reply:
Thank you. Finally I understood. The point is "bubbling'. The event bubbles from "img" through "a". Also I forgotten about features "closest'.